src/settings/FileExtensions/FileExtensions.js

```diff
@@ -106,258 +79,67 @@ export class FileExtensions extends Component {
-      areFileExtensionsResetting: false,
-    });
-  };
-
-  restoreDefaultFileExtensions = async () => {
-    const { areFileExtensionsResetting } = this.state;
-
-    if (areFileExtensionsResetting) {
-      return;
-    }
-
-    try {
-      const {
-        match: { path },
-        mutator: { restoreDefaultFileExtensions },
-      } = this.props;
-
-      this.setState({ areFileExtensionsResetting: true });
-      await restoreDefaultFileExtensions.POST({});
-
-      this.transitionToParams({
-        _path: `${path}/view`,
-        layer: null,
-      });
-
-      this.hideResetFileExtensionsToDefaultsModal();
-    } catch (error) {
-      this.hideResetFileExtensionsToDefaultsModal();
-
-      const message = (
-        <FormattedMessage
-          id="ui-data-import.settings.fileExtension.reset.error.network"
-          values={{ description: <FormattedMessage id="ui-data-import.communicationProblem" /> }}
-        />
-      );
-
-      this.calloutRef.current.sendCallout({
-        type: 'error',
-        message,
-      });
-    }
+  static defaultProps = {
+    showSingleResult: true,
+    ENTITY_KEY: ENTITY_KEYS.FILE_EXTENSIONS,
+    actionMenuItems: ['restoreDefaults'],
+    visibleColumns: [
+      'extension',
+      'importBlocked',
+      'dataTypes',
+      'updated',
+      'updatedBy',
+    ],
+    columnWidths: {
+      extension: 100,
+      dataTypes: 150,
+      updated: 150,
+      updatedBy: 250,
+    },
+    initialValues: {
+      importBlocked: false,
+      description: '',
+      extension: '',
+      dataTypes: [],
+    },
+    EntityView: ViewFileExtension,
+    EntityForm: FileExtensionForm,
   };
 
-  getDeleteRecordSuccessfulMessage(record) {
-    return (
-      <FormattedMessage
-        id="ui-data-import.settings.fileExtension.action.success"
-        values={{
-          extension: record.extension,
-          action: (
-            <FormattedMessage
-              id="ui-data-import.deleted"
-              tagName="strong"
-            />
-          ),
-        }}
-      />
-    );
-  }
-
-  getDeleteRecordErrorMessage(record) {
-    return (
-      <FormattedMessage
-        id="ui-data-import.settings.fileExtension.action.error"
-        values={{
-          extension: record.extension,
-          action: (
-            <FormattedMessage
-              id="ui-data-import.deleted"
-              tagName="strong"
-            />
-          ),
-        }}
-      />
-    );
-  }
-
-  getRecordName(record) {
-    return record.extension;
-  }
-
-  onUpdateRecordError = createUpdateRecordErrorMessage({
-    getRecordName: this.getRecordName,
-    calloutRef: this.calloutRef,
-  });
-
-  onDeleteSuccess = createDeleteCallout({
-    getMessage: this.getDeleteRecordSuccessfulMessage,
-    calloutRef: this.calloutRef,
-  });
-
-  onDeleteError = createDeleteCallout({
-    getMessage: this.getDeleteRecordErrorMessage,
-    calloutRef: this.calloutRef,
-    type: 'error',
+  renderHeaders = (intl, ...params) => ({
```

Comment:
Seems like params is unused here.

Reply:
Yes but I've left it for consistency with such methods in all the other profiles

Reply:
I propose to remove then that stuff and wait for the next story.